### ISSUE TITLE
Missing libraries for compile some projects

### ISSUE THREAD

**Issue by illera88** [beam-mw/beam#2]
Hi,

I've realized that there are some missing libraries to be linked at the time to compile the next projects:
- beam-node
- node_1_test
- node_test
- wallet_db_test
- offline
- beam-wallet

libcrypto.lib (from OpenSSL) and Crypt32.lib should be added to the linker for them to compile and link properly. 

EDIT: I guess the problem comes from https://github.com/beam-mw/beam/blob/master/sqlite/CMakeLists.txt#L32. I'm setting the path for libssl and libcrypto instead of using the recommended openSSL sources you guys mention in the readme:
![2018-09-07 17_02_34-cmake 3 11 2 - c__users_alberto garcia_documents_code_beam_build_x64](https://user-images.githubusercontent.com/5193128/45247879-e24d1400-b2bf-11e8-893b-515dda52c427.png)

**Comment by nesbox** [comment]
Hi,
We are using these prebuilt binaries for OpenSSL https://slproweb.com/products/Win32OpenSSL.html (especially Win64 OpenSSL v1.1.0i) to build the project, and CMake finds it if you set `OPENSSL_ROOT_DIR` system variable.
Where did you get your OpenSSL lib?
Thank you.

**Comment by nesbox** [comment]
Please try to remove all these dlls and add QT5 and Boost paths to the _Path Environment Variable_
`...\libs\qt511\5.11.1\msvc2017_64\bin`
`...\libs\boost_1_68_0\lib64-msvc-14.1`
Also, could you please start Beam using the debugger in Visual Studio to understand the problem?
Thank you.

**Comment by Toninoso** [comment]
I would suggest that, for Windows builds, it would be best to have a documentation similar to one here:
[bitcoin msvc build readme](https://github.com/bitcoin/bitcoin/blob/master/build_msvc/README.md)
I apologize if there already is similar page and I haven't seen it.

I am still working on building various dependencies, not knowing yet if they should built as static libs or no - that's why I think that instructions above would be useful, that is, something like:

> PS >.\vcpkg install boost:x64-windows-static `
>     libevent:x64-windows-static `
>     openssl:x64-windows-static `
>     zeromq:x64-windows-static `
>     berkeleydb:x64-windows-static `
>     secp256k1:x64-windows-static `
>     leveldb:x64-windows-static
 
Good thing is that with the latest VS2017 we don't need to generate MSVC vxproj and sln files, CMake should be suffincient. Still working on it, however.

**Comment by Toninoso** [comment]
At this point I have built all programs and libraries that don't require QT5, using VS2017 CMake support and building all the dependencies via vcpkg.

Here are the steps:
- clone and build vcpkg
- build boost using vcpkg
- build openssl using vcpkg
- make the results visible to Beam CMake  via VS2017 CMakeSettingsjson file - needs this to be added:
  "variables": [
        {
          "name": "CMAKE_TOOLCHAIN_FILE",
          "value": "<myvcpkgpath>\\scripts\\buildsystems\\vcpkg.cmake"
        }
      ]

Wallet project needed Windows bcrypt library (actually boost that it uses requires it) and was failing with unresolved externals. I had to add this to wallet CMakeLists.txt:

if(MSVC)
	target_link_libraries(wallet bcrypt)
endif()

Issue is mentioned here: [bcrypt issue](https://github.com/boostorg/uuid/issues/68) 

Unfortunately vcpkg builds QT version 5.9 and this is giving me troubles building QT dependent projects.
Still working on it.
